src/main/src/main/java/org/geoserver/catalog/WMSLayerInfo.java

```diff
@@ -54,4 +55,10 @@
     public Set<StyleInfo> getStyles();
 
     public StyleInfo getDefaultStyle();
+
+    public boolean isRespectMetadataBBox();
+
+    public void setRespectMetadataBBox(boolean respectMetadataBBox);
+
+    public boolean checkEnvelopOverLapWithNativeBounds(ReferencedEnvelope requestEnevelope);
```

Comment:
Nope, these are meant to be just be configuration carriers, no logic inside please.

Reply:
understood will move them out

Reply:
A method in GetMap works fine.